src/panels/lovelace/editor/config-elements/glance-card/hui-glance-entity-editor.ts

```diff
@@ -0,0 +1,150 @@
+  protected entities?: ConfigEntity[];
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      entities: {},
+    };
+  }
+
+  protected renderOptions(
+    entityConf: ConfigEntity,
+    index: number
+  ): TemplateResult {
+    const glanceOptions = new GlanceOptions(entityConf);
+
+    return html`
+      <paper-toggle-button>Options</paper-toggle-button>
+      <div class="options" .index="${index}">
+        <paper-input
+          label="Name"
+          .value="${glanceOptions.name}"
+          .configValue="${"name"}"
+          @value-changed="${this._optionChanged}"
+        ></paper-input>
+        <paper-input
+          label="Icon"
+          .value="${glanceOptions.icon}"
+          .configValue="${"icon"}"
+          @value-changed="${this._optionChanged}"
+        ></paper-input>
+        <paper-dropdown-menu label="Toggle Action">
+          <paper-listbox
+            slot="dropdown-content"
+            .selected="${glanceOptions.tap_action}"
+            attr-for-selected="action"
+            .configValue="${"tap_action"}"
+            @selected-item-changed="${this._optionChanged}"
+          >
+            <paper-item action="more-info">More Info Dialog</paper-item>
+            <paper-item action="toggle">Toggle</paper-item>
+            <paper-item action="call-service">Call Service</paper-item>
+          </paper-listbox>
+        </paper-dropdown-menu>
+        <paper-dropdown-menu label="Hold Action">
+          <paper-listbox
+            slot="dropdown-content"
+            .selected="${glanceOptions.hold_action}"
+            attr-for-selected="action"
+            .configValue="${"hold_action"}"
+            @selected-item-changed="${this._optionChanged}"
+          >
+            <paper-item action="">No Action</paper-item>
+            <paper-item action="more-info">More Info Dialog</paper-item>
+            <paper-item action="toggle">Toggle</paper-item>
+            <paper-item action="call-service">Call Service</paper-item>
+          </paper-listbox>
+        </paper-dropdown-menu>
+        ${
+          glanceOptions.hold_action === "call-service" ||
+          glanceOptions.tap_action === "call-service"
+            ? html`
+                <paper-input
+                  label="Service"
+                  .value="${glanceOptions.service}"
+                  .configValue="${"service"}"
+                  @value-changed="${this._optionChanged}"
+                ></paper-input>
+              `
+            : ""
+        }
+      </div>
+    `;
+  }
+
+  protected _optionChanged(ev: Event): void {
+    const target = ev.target! as EditorTarget;
+    const value = target.value || target.selected;
+    const index = target.parentElement!.index!;
+    const entity = this.entities![index];
+    const glanceOptions = new GlanceOptions(entity);
+
+    if (glanceOptions[target.configValue!] === value) {
+      return;
+    }
+
+    super._optionChanged(ev);
```

Comment:
This seems like it might become a bit of a complicated flow of data. Why is this needed?

Reply:
The function in here checks the filter. and the super function will send the changed event

Reply:
I think that we can't know that the super event handler can always handle what we have rendered in options, we shouldn't code ourselves into a corner just yet. Only think about creating an abstraction for a pattern after having done 3 examples without an abstraction but with copy paste. Only then can we start to see if there is a pattern.

I would suggest that instead of `super._optionChanged(ev)`, we make something like `this._commitChangedOptions({[target.configValue]: value})`